Maintenance

greg0ire opened the pull request.

greg0ire committed:
Remove unneeded invocation

greg0ire commented:
:thinking: I'm wondering if I need to install the mongodb extension to fix PHPStan, cc @alcaeus @franmomu

EDIT: doing so locally doesn't fix the issue.

franmomu commented:
> 🤔 I'm wondering if I need to install the mongodb extension to fix PHPStan, cc @alcaeus @franmomu
> 
> EDIT: doing so locally doesn't fix the issue.

could be something related to the PHP version set in 

https://github.com/doctrine/data-fixtures/blob/4892c74dd04a3ec7404b9586272d722325905c58/phpstan.neon.dist#L2

which is different from the PHP version running the job? 

`doctrine/mongodb-odm` 2.5 only supports PHP >= 7.4

derrabus commented:
Indeed. In other Doctrine projects, we've changed this setting to the highest supported PHP version. We should do the same here.

greg0ire commented:
Ah well spotted @franmomu , that does fix the issue. I wouldn't have thought of checking this, thanks a lot!

greg0ire committed:
Analyse project with the highest supported PHP version

greg0ire committed:
Upgrade to the latest coding standard

greg0ire committed:
Upgrade to Psalm 5

New versions warn that findUnusedBaselineEntry and findUnusedCode will
default to true in Psalm 6, and that we should set them explicitly.
Since this is a library, public APIs are flagged as unused, which is why
I am setting findUnusedCode to false.

greg0ire committed:
Upgrade to PHPUnit 10

greg0ire commented on phpunit.xml.dist:
```diff
@@ -12,9 +12,9 @@
 
-    <filter>
-        <whitelist>
+    <coverage>
```
I guess this means we only get coverage for PHP >= 7.3

derrabus replied:
Probably. FWIW, [nobody](https://packagist.org/packages/doctrine/data-fixtures/php-stats#1.6)'s using this package on PHP 7.2/7.3 anymore anyway.

greg0ire replied:
Ok then it's fine, and we should drop support for those in the next minor then.
